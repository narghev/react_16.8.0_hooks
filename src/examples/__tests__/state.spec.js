import React from 'react';
import State from '../state';
import { mount } from 'enzyme';

describe('Counter component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(<State />);
  });

  it('Renders without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Renders correct children', () => {
    expect(wrapper.find('p')).toHaveLength(1);
    expect(wrapper.find('button')).toHaveLength(1);
  });

  it('Increases the value of p when clicking on button', () => {
    const p = wrapper.find('p');
    const oldValue = Number(p.text());
    wrapper.find('button').simulate('click');
    const newValue = Number(p.text());
    expect(newValue).toEqual(oldValue + 1);
  });
});