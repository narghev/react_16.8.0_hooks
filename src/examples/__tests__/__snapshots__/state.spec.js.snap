// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component Renders without crashing 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <State />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {},
    "ref": null,
    "rendered": Object {
      "instance": <div>
        <p>
          1
        </p>
        <button>
          Increase
        </button>
      </div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            1
          </p>,
          <button
            onClick={[Function]}
          >
            Increase
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <p>
            1
          </p>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 1,
          },
          "ref": null,
          "rendered": Array [
            1,
          ],
          "type": "p",
        },
        Object {
          "instance": <button>
            Increase
          </button>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Increase",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "Increase",
          ],
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": Object {
        "instance": <div>
          <p>
            1
          </p>
          <button>
            Increase
          </button>
        </div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              1
            </p>,
            <button
              onClick={[Function]}
            >
              Increase
            </button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <p>
              1
            </p>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 1,
            },
            "ref": null,
            "rendered": Array [
              1,
            ],
            "type": "p",
          },
          Object {
            "instance": <button>
              Increase
            </button>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Increase",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              "Increase",
            ],
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
